### PR TITLE
[CES-88] Added MySQL Flexible Server module

### PR DESCRIPTION
**Pull request by mamu0**
### List of changes

Created a new module in `infra/modules/azure_mysql_server`.

### Motivation and context

This change is required to facilitate the management and deployment of MySQL servers on Azure in a modular and flexible way. It enables the creation of a MySQL server using Infrastructure as Code (IaC), improving the maintainability and automation of the cloud environment.

Possibles tiers are:
- `test`:
  - MySQL Flexible Server (Basic)
  - Private Endpoint
  - (Optional) Alerts - as default disabled
  - (Optional) Diagnostic - as default disabled
- `standard`:
  - Same as test, but Server have HA and Backup configured 
- `premium`:
  - Same as standard, but there is a Replica DB with Private Endpoint and monitoring

The MySQL Server name will be in the form:
- `[prefix]-[env_short]-[location_short]-[domain]-mysql-[instance_number]`
Private Endpoint:
- `[prefix]-[env_short]-[location_short]-[domain]-mysql-pep-[instance_number]`

If Replica is enabled:
- `[prefix]-[env_short]-[location_short]-[domain]-mysql-replica-[instance_number]`
- `[prefix]-[env_short]-[location_short]-[domain]-mysql-pep-replica-[instance_number]`

As outputs we will obtain:

- **Private Endpoint ID**
- **MySQL** `name` and `ID`
- (If premium) replica **rivate Endpoint ID**
- (If premium) replica **MySQL** `name` and `ID`

Example:

```
data "azurerm_key_vault_secret" "ps_admin_name" {
  name         = "ps-admin-name"
  key_vault_id = data.azurerm_key_vault.kv.id
}

data "azurerm_key_vault_secret" "ps_admin_pwd" {
  name         = "ps-admin-pwd"
  key_vault_id = data.azurerm_key_vault.kv.id
}

module "mysql" {
  source = "../../modules/azure_mysql_server"

  environment = {
    prefix          = "dx"
    env_short       = "d"
    location        = "italynorth"
    app_name         = "test"
    instance_number = "01"
  }

  db_version          = "8.0.21"
  tier                = "test"
  subnet_pep_id         = "/subscription/..."
  resource_group_name = "dev"

  administrator_credentials = {
    name     = data.azurerm_key_vault_secret.ps_admin_name.value
    password = data.azurerm_key_vault_secret.ps_admin_pwd.value
  }

  tags = {
    Owner = "dx"
    ...
  }
}

```

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Other information

Ref.  [azurerm_mysql_flexible_server ](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mysql_flexible_server)

### CODE REVIEW

File: infra/modules/azure_mysql_server/main.tf [hunk 1]
```diff
@@ -0,0 +1,68 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 3.116.0"
```
Comment (Krusty93) [inline infra/modules/azure_mysql_server/main.tf:5]: Is this very high version mandatory also here?

File: infra/modules/azure_mysql_server/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,28 @@
+
+output "private_endpoint" {
+
+  value       = azurerm_private_endpoint.mysql_pe.id
+  description = "Subnet ID in which to create the MySQL Flexible Server Replica"
```
Comment (Krusty93) [inline infra/modules/azure_mysql_server/outputs.tf:5]:
```suggestion
  description = "MySql Private Endpoint resource id"
```

File: infra/modules/azure_mysql_server/main.tf (hunk 1 above)
Comment (mamu0) [reply infra/modules/azure_mysql_server/main.tf]: actually no, now I installed `3.111` which doesn't seem to cause any problems

**Comment by Krusty93** [comment]
A confirmation - shall we discard? @gunzip

**Comment by gunzip** [comment]
I'd move this into io-infra folder

**Comment by mamu0** [comment]
> I'd move this into io-infra folder

Good, then I'll move it into `io-infra`

**Pull request closed without merging**